lib/common/repository/auth_repository.dart

```diff
@@ -0,0 +1,78 @@
+import 'dart:developer';
+
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:fluttercon/common/data/models/models.dart';
+import 'package:fluttercon/common/utils/network.dart';
+import 'package:google_sign_in/google_sign_in.dart';
+import 'package:injectable/injectable.dart';
+import 'package:logger/logger.dart';
+
+@singleton
+class AuthRepository {
+  final _networkUtil = NetworkUtil();
+
+  final FirebaseAuth _auth = FirebaseAuth.instance;
+
+  final GoogleSignIn _googleSignIn = GoogleSignIn(
+    scopes: [
+      'profile',
+      'email',
+    ],
+  );
+
+  Future<String> signInWithGoogle() async {
+    try {
+      final googleSignInAccount = await _googleSignIn.signIn();
+      final googleSignInAuthentication =
+          await googleSignInAccount?.authentication;
+
+      final AuthCredential credential = GoogleAuthProvider.credential(
+        idToken: googleSignInAuthentication?.idToken,
+        accessToken: googleSignInAuthentication?.accessToken,
+      );
+
+      final authResult = await _auth.signInWithCredential(credential);
+
+      final user = authResult.user;
+
+      if (user != null) {
+        assert(!user.isAnonymous, 'User must not be anonymous');
+        return Future.value(authResult.credential?.accessToken);
```

Comment:
Curious, why do we use the Future.value and not the result directly?

Reply:
This is so because the function `Future<String> signInWithGoogle()` expects a future to be returned and this is the hacky way to return the future with that string inside. 🥶